### PR TITLE
Add tests for sendCustomMessage() for container (iframe/wc)

### LINKED ISSUE #3845: Test sendCustomMessage for LuigiContainer
This issue is part of https://github.com/SAP/luigi/issues/3757

### PR DESCRIPTION
**Pull request by VincentUllal**
Changes proposed in this pull request:

added tests for sendCustomMessage() for Container
- webcomponent
- iframe

**Related issue(s)**

 Resolves #3845

### CODE REVIEW

File: container/cypress/e2e/test-app/iframe/iframe-container.cy.js [hunk 1]
```diff
@@ -23,6 +23,32 @@ describe('Iframe Container Test', () => {
...
 
+  it('sendCustomMessage', () => {
+    cy.get(containerSelector)
+      .shadow()
+      .get('iframe')
+      .then(iframe => {
+        const $body = iframe.contents().find('body')
+        cy.wrap($body)
+          .find('#content')
+          .should('have.text', ' ');
```
Comment (ndricimrr) [inline container/cypress/e2e/test-app/iframe/iframe-container.cy.js:34]: whats the idea behind this empty string text check ?

File: container/cypress/e2e/test-app/iframe/iframe-container.cy.js [hunk 2]
```diff
@@ -23,6 +23,32 @@ describe('Iframe Container Test', () => {
...
+      });
+
+    cy.get('#btn-1')
+      .click()
+      .then(() => {
+        cy.get(containerSelector)
```
Comment (ndricimrr) [inline container/cypress/e2e/test-app/iframe/iframe-container.cy.js:40]: nested **cy.get('#btn-1')** > **cy.get(containerSelector)** is not a good approach. 

You can do it step by step. Don't have to nest them like this.

File: container/cypress/e2e/test-app/wc/wc-container.cy.js [hunk 3]
```diff
@@ -131,5 +131,20 @@ describe('Web Container Test', () => {
...
+        .find('#customMessageDiv')
+        .should('have.text', 'Received Custom Message: ');
+
+      cy.get('#sendCustomMessageBtn')
+        .click()
+        .then(() => {
+          cy.get(containerSelector)
+            .shadow()
+            .find('#customMessageDiv')
+            .should('have.text', 'Received Custom Message: cool custom Message');
```
Comment (ndricimrr) [inline container/cypress/e2e/test-app/wc/wc-container.cy.js:146]: why are there 2 **should('have.text)** tests for the same thing here . 1 is enough. I think scenario is. 
1. Click button to send message from  outside, and 
2. check if message received inside container mf

File: container/test-app/wc/clientAPI.html [hunk 4]
```diff
@@ -33,9 +38,6 @@ <h3>
         ></luigi-container>
       </div>
 
-      <button id="init-button" style="margin: 25px 0 15px">
-        container.init() to initialize defered container
-      </button>
```
Comment (ndricimrr) [inline container/test-app/wc/clientAPI.html:38]: should keep this button where it was. as it belongs to the container right below it. the one above doesnt relate to **init** scenario

File: container/test-app/wc/clientAPI.html [hunk 5]
```diff
@@ -10,11 +10,16 @@ <h3>
...
+    <button id="init-button" style="margin: 25px 0 15px">
+      container.init() to initialize defered container
+    </button>
+    <button id="sendCustomMessageBtn" style="margin: 25px 0 15px">sendCustomMessage</button>
 
     <div style="height: fit-content;">
       <div style="border: 1px solid blue">
         <!-- Luigi Container to test general functionalities, params etc-->
         <luigi-container
+          id="test-id"
```
Comment (ndricimrr) [inline container/test-app/wc/clientAPI.html:22]: you already have **data-test-id**, you can fetch the container using that on Cypress. If you do need it, give it a more specific id names so its distinguishable from other containers in this file.

**Review by ndricimrr — COMMENTED** [review]
some comments on the tests

File: container/cypress/e2e/test-app/iframe/iframe-container.cy.js (hunk 1 above)
Comment (VincentUllal) [reply container/cypress/e2e/test-app/iframe/iframe-container.cy.js:34]: I wanted to check the states before and after sendCustomMessage() but you're right, the first check isn't necessary :+1:

File: container/test-app/wc/clientAPI.html (hunk 5 above)
Comment (VincentUllal) [reply container/test-app/wc/clientAPI.html:22]: Ok I'll remove it, I'll use the data-test-id

File: container/cypress/e2e/test-app/wc/wc-container.cy.js [hunk 6]
```diff
@@ -131,5 +131,15 @@ describe('Web Container Test', () => {
             });
         });
     });
+    it('sendCustomMessage', () => {
+      cy.get('#sendCustomMessageBtn')
+        .click()
```
Comment (ndricimrr) [inline container/cypress/e2e/test-app/wc/wc-container.cy.js:136]: try making a chained **.then** call just like in the updateContext test above. 🤔

**Review by ndricimrr — APPROVED** [review]
looks good

**Pull request merged**